nwb_conversion_tools/conversion_tools.py

```diff
@@ -0,0 +1,57 @@
+"""Authors: Cody Baker."""
+from pathlib import Path
+import spikeextractors as se
+
+
+def save_si_object(si_object, output_folder, cache_raw=False, include_properties=True, include_features=False):
+    """
+    Save an arbitrary SI object to a temprary location for NWB conversion.
+
+    Parameters
+    ----------
+    si_object: RecordingExtractor or SortingExtractor
+        The extractor to be saved.
+    output_folder: str or Path
+        The folder where the object is saved.
+    cache_raw: bool
+        If True, the Extractor is cached to a binary file (not recommended for RecordingExtractor objects)
+        (default False).
+    include_properties: bool
+        If True, properties (channel or unit) are saved (default True).
+    include_features: bool
+        If True, spike features are saved (default False)
+
+    Returns
+    -------
+    spikeinterface_folder: str
+        The output spikeinterface folder
+    """
+    spikeinterface_folder = Path(output_folder) / "spikeinterface"
+    spikeinterface_folder.mkdir(parents=True, exist_ok=True)
+
+    if isinstance(si_object, se.RecordingExtractor):
+        if not si_object.is_dumpable or cache_raw:
+            cache = se.CacheRecordingExtractor(si_object, save_path=spikeinterface_folder / "raw.dat")
+        else:
+            cache = si_object
+
+    elif isinstance(si_object, se.SortingExtractor):  # TODO: asssumes a single extractor at the moment (not  iterable)
+        if not si_object.is_dumpable or cache_raw:
+            cache = se.CacheSortingExtractor(si_object, spikeinterface_folder / "sorting.npz")
+        else:
+            cache = si_object
+    else:
+        raise ValueError("The 'si_object' argument shoulde be a SpikeInterface Extractor!")
+
+    json_file = spikeinterface_folder / "raw.json"
+    pkl_file = spikeinterface_folder / "raw.pkl"
```

Comment:
@alejoe91 Just to clarify, if we dump both a recording and sorting object to these files using this method, the single .json and .pkl files will contain all the information for both extractor objects? (since the name of these files is hard-coded and not different per object)

Reply:
Or does the cache.dump_to_* otherwise automatically make the file name unique per cache?

Reply:
I think we should dump the json (we should move it before the caching option) because that keeps all the information, while after caching it's resetted to the new bin file.